psi4/src/psi4/libscf_solver/hf.cc

```diff
@@ -720,23 +724,34 @@ void HF::find_occupation()
     } else {
         std::vector<std::pair<double, int> > pairs_a;
         std::vector<std::pair<double, int> > pairs_b;
-        for (int h=0; h<epsilon_a_->nirrep(); ++h) {
-            for (int i=0; i<epsilon_a_->dimpi()[h]; ++i)
+        for (int h = 0; h < epsilon_a_->nirrep(); ++h) {
+            for (int i = 0; i < epsilon_a_->dimpi()[h]; ++i){
                 pairs_a.push_back(std::make_pair(epsilon_a_->get(h, i), h));
+            }
         }
-        for (int h=0; h<epsilon_b_->nirrep(); ++h) {
-            for (int i=0; i<epsilon_b_->dimpi()[h]; ++i)
-                pairs_b.push_back(std::make_pair(epsilon_b_->get(h, i), h));
+        sort(pairs_a.begin(), pairs_a.end());
+
+        // Do we need to sort beta?
```

Comment:
The only issue I can see is with broken symmetry computations. However, I am convincing myself this is not really an issue. If ran in C1 symmetry, the lack of beta sorting should not make any difference. On the other hand, I don't think Psi4 can do broken symmetry computations with non-C1 groups and use different alpha and beta occupation vectors. This is necessary to represent determinants like:

`|(core)^2 (4a' alpha) (2a" beta)>` (That's what we do in OCDFT, and there we have tricks to avoid psi4 sorting the MOs).

So in the end this might not be an issue at all.